src/commtrack.js

```diff
@@ -263,61 +279,85 @@ define([
-                    xpathType: "generic",
-                    help: 'A reference to an integer question in this form.',
-                },
-                relevantAttr: {
-                    visibility: 'visible',
-                    presence: 'optional',
-                    widget: widgets.xPath,
-                    xpathType: "bool",
-                    lstring: 'Display Condition'
-                },
                 requiredAttr: { presence: "notallowed" },
                 constraintAttr: { presence : "notallowed" },
                 calculateAttr: { presence: "notallowed" }
             }
-        };
+        }),
+        dispenseMugOptions = util.extend(transferMugOptions, {
+            typeName: 'Dispense',
+            icon: 'fcc fcc-fd-door',
+            spec: {
+                src: {
+                    validationFunc: function (mug) {
+                        if (mug.p.src.value) {
+                            return 'pass';
+                        }
+                        return 'Receive must have a Source Case.';
+                    },
+                },
+                dest: { presence: "notallowed" },
+            }
+        }),
+        receiveMugOptions = util.extend(transferMugOptions, {
+            typeName: 'Receive',
+            icon: 'fcc fcc-fd-inbox',
+            spec: {
+                src: { presence: "notallowed" },
+                dest: {
+                    validationFunc: function (mug) {
+                        if (mug.p.dest.value) {
+                            return 'pass';
+                        }
+                        return 'Receive must have a Destination Case.';
+                    },
+                },
+            }
+        });
 
     $.vellum.plugin("commtrack", {}, {
         getAdvancedQuestions: function () {
-            return this.__callOld().concat(["Balance", "Transfer"]);
+            return this.__callOld().concat([
+                "Balance",
+                "Transfer",
+                "Dispense",
+                "Receive"
+            ]);
         },
         getMugTypes: function () {
             var types = this.__callOld();
-            types.normal.Balance = util.extend(
-                mugs.defaultOptions, baseTransactionOptions, balanceMugOptions);
-            types.normal.Transfer = util.extend(
-                mugs.defaultOptions, baseTransactionOptions, transferMugOptions);
+            types.normal.Balance = balanceMugOptions;
+            types.normal.Transfer = transferMugOptions;
+            types.normal.Dispense = dispenseMugOptions;
+            types.normal.Receive = receiveMugOptions;
             return types;
         },
+        parseDataElement: function (form, el, parentMug, role) {
+            var tag = el.nodeName;
+            if (!role && (tag === "transfer" || tag === "balance") &&
+                    $(el).attr("xmlns") === LEDGER_XMLNS) {
+                if (tag === "transfer") {
```

Comment:
there are even tests for it [here](https://github.com/dimagi/Vellum/commit/d57ea083c026656b2b9e3ac0ee1c07449840c772#diff-ff401b6488cc00a539c9523227c94c15R314) and [here](https://github.com/dimagi/Vellum/commit/d57ea083c026656b2b9e3ac0ee1c07449840c772#diff-905dcd5f60ed0d9e12b350401d356a0aR416)